appservice/src/tree/DeploymentsTreeItem.ts

```diff
@@ -0,0 +1,80 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { SiteConfig } from 'azure-arm-website/lib/models';
+import * as path from 'path';
+import { AzureParentTreeItem, DialogResponses, GenericTreeItem, IActionContext } from 'vscode-azureextensionui';
+import KuduClient from 'vscode-azurekudu';
+import { DeployResult } from 'vscode-azurekudu/lib/models';
+import { editScmType } from '../editScmType';
+import { ext } from '../extensionVariables';
+import { getKuduClient } from '../getKuduClient';
+import { ScmType } from '../ScmType';
+import { DeploymentTreeItem } from './DeploymentTreeItem';
+import { ISiteTreeRoot } from './ISiteTreeRoot';
+
+export class DeploymentsTreeItem extends AzureParentTreeItem<ISiteTreeRoot> {
+    public static contextValueConnected: string = 'deploymentsConnected';
+    public static contextValueUnconnected: string = 'deploymentsUnconnected';
+    public contextValue: string = DeploymentsTreeItem.contextValueUnconnected;
+    public parent: AzureParentTreeItem<ISiteTreeRoot>;
+    public readonly label: string = 'Deployments';
+    public readonly childTypeLabel: string = 'Deployment';
+
+    public constructor(parent: AzureParentTreeItem<ISiteTreeRoot>, siteConfig: SiteConfig) {
+        super(parent);
+        this.contextValue = siteConfig.scmType === ScmType.None ? DeploymentsTreeItem.contextValueUnconnected : DeploymentsTreeItem.contextValueConnected;
+    }
+
+    public get iconPath(): { light: string, dark: string } {
+        return {
+            light: path.join(__filename, '..', '..', '..', 'resources', 'light', 'Deployments_x16.svg'),
+            dark: path.join(__filename, '..', '..', '..', 'resources', 'dark', 'Deployments_x16.svg')
+        };
+    }
+
+    public hasMoreChildrenImpl(): boolean {
+        return false;
+    }
+
+    public async loadMoreChildrenImpl(_clearCache: boolean): Promise<DeploymentTreeItem[] | GenericTreeItem<ISiteTreeRoot>[]> {
+        const siteConfig: SiteConfig = await this.root.client.getSiteConfig();
+        if (siteConfig.scmType === ScmType.GitHub || siteConfig.scmType === ScmType.LocalGit) {
+            const kuduClient: KuduClient = await getKuduClient(this.root.client);
+            const deployments: DeployResult[] = await kuduClient.deployment.getDeployResults();
+            return deployments.map((deployResult: DeployResult) => {
+                return new DeploymentTreeItem(this, deployResult, kuduClient);
```

Comment:
Probably worth leveraging `createTreeItemsWithErrorHandling` so that one invalid deployResult doesn't prevent the user from viewing all valid deployResults (and because I just don't trust kudu to always return valid stuff)